### PR TITLE
Improve sequence diagrams

### PR DESCRIPTION
**Pull request by simonteozw**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -324,6 +324,11 @@ In this section, we will outline the key operations of the Module Manager, namel
 * `AddModuleCommand`
 * `DeleteModuleCommand`
 * `EditModuleCommand`
+
+Before we begin, it is important for us to define 1 key term: **association**:
+
+:information_source: A lesson/task/contact is associated with a module if its **code** field is equal to the
+ module's **code**.
```
Comment (UncleGrandpa925) [inline docs/DeveloperGuide.md]: Actually "a contact is associated with a module if it contains that module'**code** as one of its tags"

**Comment by UncleGrandpa925** [comment]
Btw some additional reviews: 
image

2 and 4 I think should be merged into 1 and 3

**Comment by UncleGrandpa925** [comment]
image

I think the call to Task here seems to be too low-level detail?

**Comment by UncleGrandpa925** [comment]
image

I think you should mention "editedTask has a valid code or no code"

**Comment by UncleGrandpa925** [comment]
image
Just asking: should we show the call to TrackIter here?

**Comment by simonteozw** [comment]
> image
> Just asking: should we show the call to TrackIter here?

It doesnt call TrackIter it just calls filteredList.setPredicate()

**Review by UncleGrandpa925 — APPROVED** [review]

**Pull request merged**